### ISSUE TITLE
Check disposal handling in GIFs

### ISSUE THREAD

**Issue by TWiStErRob** [bumptech/glide#1094]
While building #1093 I noted that probably the `DISPOSAL_PREVIOUS` handling should be parametrized similarly as well, but this needs confirmation by a GIF that does this and fails to render correctly, code is probably:

``` java
previousImage.getPixels(dest, topLeft, width, ix, iy, iw, ih);
```

based on

>  iv) Disposal Method - Indicates the way in which the graphic is to be treated after being displayed.
> 3 - Restore to previous. The decoder is required to restore **the area overwritten by the graphic** with what was there prior to rendering the graphic. -- [GIF89a](https://www.w3.org/Graphics/GIF/spec-gif89a.txt), emphasis mine

**Comment by TWiStErRob** [comment]
A good source to double-check this is http://www.imagemagick.org/Usage/anim_basics/

**Comment by TWiStErRob** [comment]
Creating a branch locally where every current PR related to GIF is merged:
 * ![](http://www.imagemagick.org/Usage/anim_basics/anim_bgnd.gif)  
leaves a white trace instead of transparent
 * ![](http://www.imagemagick.org/Usage/anim_basics/bunny_anim.gif)  
frames drawn over each other instead of after another

**Comment by TWiStErRob** [comment]
I made progress so the bunny is displayed correctly.
I also applied my hunch in the initial comment and it seems everything is working correctly.
More tests need to be done to confirm that partial "dispose to previous" is correct.
If anyone wants to picks this up, here are the changes: https://github.com/TWiStErRob/glide/commit/c8c331ed60b0a8e5133aa4c0e29780c45afdde17

---

The other image seems to be correct according to my reading of the spec, but it seems that browsers and other image software agree on "background" always meaning transparent and not `bgColor`.

**Comment by sjudd** [comment]
Was this fixed by #2521?

**Comment by michaeimm** [comment]
#2521 only fixed that bunny GIF.

**Comment by michaeimm** [comment]
This seems be fixed by #3002 .

**Comment by sjudd** [comment]
Great thanks!
